integration/ipam.go

```diff
@@ -0,0 +1,118 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"net"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	defaultCIDRMask   = 24
+	totalBitsLength   = 32
+	privateSubnetMask = 25
+	publicSubnetMask  = 25
+	networkCIDR       = "10.1.0.0/16"
+)
+
+type awsVPC struct {
+	NetworkCIDR       string
+	PrivateSubnetCIDR string
+	PublicSubnetCIDR  string
+}
+
+func newAWSVPCBlock(client AWSClient) (awsVPC, error) {
+	cidrMask := net.CIDRMask(defaultCIDRMask, totalBitsLength)
+	existingSubnets, err := listSubnets(client)
+	if err != nil {
+		return awsVPC{}, microerror.Mask(err)
+	}
+	cidr, err := newSubnet(cidrMask, existingSubnets)
+	if err != nil {
+		return awsVPC{}, microerror.Mask(err)
+	}
+
+	// Configure private and public subnets as subnets of the new subnet.
+	privateSubnetMask := net.CIDRMask(privateSubnetMask, totalBitsLength)
+	privateSubnetCIDR, err := ipam.Free(cidr, privateSubnetMask, nil)
+	if err != nil {
+		return awsVPC{}, microerror.Mask(err)
+	}
+
+	publicSubnetMask := net.CIDRMask(publicSubnetMask, totalBitsLength)
+	publicSubnetCIDR, err := ipam.Free(cidr, publicSubnetMask, []net.IPNet{privateSubnetCIDR})
```

Comment:
I'm not super clear on ipam stuff, but doesn't that always return privateSubnetCIDR?

Reply:
According to IPAM's `Free` https://github.com/giantswarm/ipam/blob/master/ipam.go#L131 it would return a network within `cidr`, with the given mask and not overlapping with `privateSubnetCIDR`, is that correct?